explainer.md

```diff
@@ -53,20 +52,32 @@ See individual sections for more details on these characteristics.
 
 ## Type definitions
 
-The central part of the Typed Objects specification are *type definition objects*, generally called *type definitions* for short. Type definitions describe the fixed structure of a value.
+The central part of the Typed Objects specification are *type definition objects*, generally called *type definitions* for short. Type definitions describe the fixed structure of an instance in memory, specifying a struct type's fields' types.
 
-### Value type definitions
+A type definition has the internal methods `[[Read]]` and `[[Write]]`, which convert between `Value` and internal representations for the type. `[[Write]]` additionally performs a type-specific type check.
```

Comment:
I think we need to be careful to keep the concepts of *type definition* and *value type* distinct; in particular *type definition*s do not have `[[Read]]` or `[[Write]]` internal methods; rather type definitions *contain* value types and it is the *value types* that exclusively have the `[[Read]]` and `[[Write]]` methods that get invoked by the `[[Get]]` and `[[Set]]` internal methods of instances.

In particular, I'd pull the "Value Types" subsection out of this "Type Definitions" section and make "Value Types" into a full section.  There's an annoying sequencing constraint (Value Types logically comes first, but you kinda want to motivate it by introducing Type Definitions first), but I suppose it could be solved either way.

Reply:
As a comparison, check out the "Basic Approach" section in my WIP GC JS API (is that enough acronyms for you?) draft:
https://github.com/WebAssembly/gc/blob/add-js-api/proposals/gc/MVP-JS.md

Reply:
Good call, this was indeed just wrong. I restructured things quite a bit, and think it's much better now.